src/core/binding.c

```diff
@@ -382,15 +382,18 @@ _Success_(return != NULL)
 QUIC_LISTENER*
 QuicBindingGetListener(
     _In_ QUIC_BINDING* Binding,
+    _In_opt_ QUIC_CONNECTION* Connection,
     _Inout_ QUIC_NEW_CONNECTION_INFO* Info
     )
 {
+    UNREFERENCED_PARAMETER(Connection);
     QUIC_LISTENER* Listener = NULL;
 
     const QUIC_ADDR* Addr = Info->LocalAddress;
     const QUIC_ADDRESS_FAMILY Family = QuicAddrGetFamily(Addr);
 
     BOOLEAN FailedAlpnMatch = FALSE;
+    BOOLEAN FailedAddrMatch = FALSE;
```

Comment:
I think this should actually default to `TRUE` and then set to `FALSE` if it passes the addr check below. The reason being is that if (somehow) no listeners are present in this loop, then that should be treated as no addr.